app/controllers/concerns/current_user.rb

```diff
@@ -38,6 +38,7 @@ def warden
 
   def unauthorized!
     Rails.logger.warn "Halted as unauthorized! threw :warden (called from #{caller.first.sub(Rails.root.to_s, '')})"
+    Samson::Hooks.fire(:unauthorized_action, current_user, controller_name, action_name)
```

Comment:
... to reduce spam / the "initial login" flow ... otherwise you won't be able to see anyone trying to mess with things, just constant spam ...